week3/src/App.jsx

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from "react";
+import "./App.css";
+import { ThemeProvider } from "styled-components";
+import GlobalStyle from "./styles/GlobalStyle";
+import theme from "./styles/theme";
+import Header from "./components/header";
+import MainPage from "./pages/CardGamePage";
+import Modal from "./components/Modal";
+import LevelButton from "./components/LevelButton";
+import { IMAGEDATA } from "./contants/images";
+
+function App() {
+  const [score, setScore] = useState(0);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [difficulty, setDifficulty] = useState("easy");
+  const [cards, setCards] = useState([]);
+  const [numPairs, setNumPairs] = useState(5); // 초기 카드 쌍 수 설정
+
+  const getMixedCardList = (numPairs) => {
+    const shuffled = [...IMAGEDATA].sort(() => 0.5 - Math.random()); // IMAGEDATA를 복사하여 사용
+    const selectedImages = shuffled.slice(0, numPairs);
+    setNumPairs(numPairs);
+    return [...selectedImages, ...selectedImages]
+      .map((card) => ({ ...card, status: false }))
+      .sort(() => 0.5 - Math.random());
+  };
+
+  // 최초 로드 시 기본 난이도의 카드 세트 로드
+  useEffect(() => {
+    setCards(getMixedCardList(5)); // Easy 모드 기본 설정
+  }, []);
+
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <GlobalStyle />
```

Comment:
p5: 최상위 쯤 위치해야하는 라이브러리의 Provider나, GlobalStyle은 보통 선호에 따라서 App.jsx / main.jsx 애 적용하게 되는데, App.jsx에 적용시키셨군요! GlobalStyle에서 font가 적용되지 않는게 저도 의문이네요, 흠.. div 위에 GlobalStyle을 위치시켜보시겠어요,,?

```suggestion
  <ThemeProvider theme={theme}>
      <GlobalStyle />
      <div>
```

Reply:
div 위에 GlobalStyle를 옮겨봤지만 폰트는 안바뀌네요 🥹 추후 왜 이런지 알게 되면 공유하겠슴니다..